Framework/DataHandling/inc/MantidDataHandling/ReadMaterial.h

```diff
@@ -0,0 +1,52 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2009 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_DATAHANDLING_ReadMaterial_H_
+#define MANTID_DATAHANDLING_ReadMaterial_H_
+
+#include "MantidKernel/MaterialBuilder.h"
+
+#include <map>
+namespace Mantid {
+namespace Kernel {
+class Material;
+}
+namespace DataHandling {
+using ValidationErrors = std::map<std::string, std::string>;
+/**
+    This class contains code for interpreting a material input for
+   SetSampleMaterial
```

Comment:
Flesh this out a little. Why is it different from `MaterialBuilder`? Because it validates the parameters...